src/engine/QueryExecutionContext.h

```diff
@@ -116,6 +118,17 @@ class QueryExecutionContext {
     updateCallback_(nlohmann::ordered_json(runtimeInformation).dump());
   }
 
+  // Getter for the global `NEXT_NEWBLANKNODEINDEX` counter.
+  [[nodiscard]] static int getNextNewBlankNodeIndex() {
+    if (NEXT_NEWBLANKNODEINDEX == std::numeric_limits<uint64_t>::max())
+        [[unlikely]] {
+      throw std::runtime_error(
+          "Counter for blank nodes overflowed. Tell the developers to reset "
+          "it.");
+    }
+    return NEXT_NEWBLANKNODEINDEX++;
+  }
```

Comment:
Unfortunately this has a race condition (there is one blank node left, two threads first run the check in line 123, 
(which for both is okay), then they both increment by one , upps overflow.
You either have to work with a `compare_exchange_weak` loop (I am sure that cppreference has an example)
or with a `std::mutex`. 
Also you could here directly also implement the function `getMultipleNewBlankNodes` that reserves several nodes (e.g. 100) at once for performance reasons.